### PR TITLE
Refactor: remove store usage in alerts helpers

### PR DESCRIPTION
**Pull request by marcinbodnar**
With this PR we are removing all imports of the `store` and using thunks instead for `alerts.js` file.

 - `dispatchWithAlert` is now `withAlert` thunk and is moved to `status` slice file.
 - `actionsPending` is now `selectActionsPending` selector.
 - `actionsPendingByPrefix ` is now `selectActionsPendingByPrefix` selector.
 - `handleClearAlert` is now a thunk and is moved to `status` slice file.

### CODE REVIEW

File: packages/frontend/src/redux/slices/status/index.js [hunk 1]
```diff
@@ -10,3 +16,18 @@ export const selectStatusMainLoader = createSelector(selectStatusSlice, (status)
 export const selectStatusLocalAlert = createSelector(selectStatusSlice, (status) => status.localAlert || {});
 
 export const selectStatusActionStatus = createSelector(selectStatusSlice, (status) => status.actionStatus || []);
```
Comment (andy-haynes) [inline packages/frontend/src/redux/slices/status/index.js:18]: I know this isn't new to this PR but shouldn't the fallback here be a `{}`? It looks like all usages of this selector rely have property access on them.

**Review by andy-haynes — CHANGES_REQUESTED** [review]
Just a few comments inline and some rebase conflicts, otherwise looking good

File: packages/frontend/src/redux/slices/status/index.js [hunk 2]
```diff
@@ -10,3 +16,18 @@ export const selectStatusMainLoader = createSelector(selectStatusSlice, (status)
...
+
+export const selectActionsPending = createSelector(
+    [getTypes, selectStatusActionStatus],
+    (types, actionStatus) => (typeof types === 'string' ? [types] : types).some((type) => actionStatus[type]?.pending)
+);
+
+export const selectActionsPendingByPrefix = createSelector(
```
[the remark this answers is not in the document]
Comment (marcinbodnar) [reply packages/frontend/src/redux/slices/status/index.js]: Done.

File: packages/frontend/src/redux/slices/status/index.js (hunk 1 above)
Comment (marcinbodnar) [reply packages/frontend/src/redux/slices/status/index.js]: You're right. Thanks. Done.

**Review by andy-haynes — APPROVED** [review]
LGTM 👍

**Pull request merged**